test/integration_test/snapshot_test.go

```diff
@@ -1180,6 +1187,479 @@ func updateRetainValueTest(t *testing.T) {
+	log.FailOnError(t, err, "Error getting volume for pvc")
+	log.InfoD("pvName for second PVC: %v", pvName2)
+
+	volume1, err := volumeDriver.InspectVolume(pvName1)
+	log.FailOnError(t, err, "Error getting inspect volume for %v", pvName1)
+	volume2, err := volumeDriver.InspectVolume(pvName2)
+	log.FailOnError(t, err, "Error getting inspect volume for %v", pvName2)
+	vol1Node, vol2Node := volume1.AttachedOn, volume2.AttachedOn
+	log.InfoD("nodes where volumes are attached: %s and %s", vol1Node, vol2Node)
+
+	nodeList, err := core.Instance().GetNodes()
+	log.FailOnError(t, err, "Error getting list of nodes")
+
+	for _, item := range nodeList.Items {
+		for _, address := range item.Status.Addresses {
+			if address.Type == "InternalIP" {
+				if address.Address == vol1Node {
+					vol1Node = item.Status.Addresses[1].Address
+				}
+				if address.Address == vol2Node {
+					vol2Node = item.Status.Addresses[1].Address
+				}
+			}
+		}
+	}
+
+	log.InfoD("node names where volumes are attached: %s and %s", vol1Node, vol2Node)
+	// Bring down portworx on the node where the PVC is attached.
+	err = core.Instance().AddLabelOnNode(vol1Node, schedops.PXServiceLabelKey, k8sServiceOperationStop)
+	log.FailOnError(t, err, "Error bringing down node")
+	err = core.Instance().AddLabelOnNode(vol2Node, schedops.PXServiceLabelKey, k8sServiceOperationStop)
+	log.FailOnError(t, err, "Error bringing down node")
+	defer func(node1, node2 string) {
+		// Bring portworx backup up on the node where the PVC is attached.
+		err = core.Instance().AddLabelOnNode(node1, schedops.PXServiceLabelKey, k8sServiceOperationStart)
+		log.FailOnError(t, err, "Error starting PX on node")
+		err = core.Instance().AddLabelOnNode(node2, schedops.PXServiceLabelKey, k8sServiceOperationStop)
+		log.FailOnError(t, err, "Error bringing down node")
+	}(vol1Node, vol2Node)
+
+	// Wait for some snapshots to go into `Error` state.
+	log.Info("Waiting for 2 minutes to make sure some snapshots get errored out")
+	time.Sleep(2 * time.Minute)
+
+	// Verify that the snapshots are in `Error` state.
+	snapshots, err := k8sextops.Instance().ListSnapshots(namespace)
+	log.FailOnError(t, err, "Error listing snapshots after bringing down the node")
+	errorCount := 0
+	errorSnapshots := make([]string, 0)
+
+	for _, snap := range snapshots.Items {
+		if snap.Status.Conditions[0].Type == crdv1.VolumeSnapshotConditionError {
+			errorCount++
+			errorSnapshots = append(errorSnapshots, snap.Metadata.Name)
+		}
+	}
+	log.Info("Number of errorsnapshots:%d", errorCount)
+	log.Info("Error snapshots list:%v", errorSnapshots)
+	if errorCount == 0 {
+		Dash.Fatal("should have more than one snapshots in Error state")
+	}
+
+	// Wait for 90 seconds more.
+	log.Info("Waiting for 90 seconds more to prevent any transient error")
+	time.Sleep(90 * time.Second)
+
+	// Verify that some of snapshots have been cleaned up.
+	snapshots, err = k8sextops.Instance().ListSnapshots(namespace)
```

Comment:
Let's add a common function to check in a volumesnapshotschedule , how many are in errored state and also deleted , also can the verify function be common?

Reply:
Done